### PR TITLE
Add Azure credentials type in configuration 

### PR DESCRIPTION
**Pull request by moukoublen**
### Summary of your changes

Based on the discussion in [this comment](https://github.com/elastic/integrations/pull/8376#discussion_r1390451815), this PR adds the Azure credentials type in the config and uses it to identify which Azure `TokenCredential` to use.

Enumerator values are on par with [this PR](https://github.com/elastic/kibana/pull/171069).

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [ ] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [ ] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

### CODE REVIEW

File: resources/providers/azurelib/auth/credentials_test.go [hunk 1]
```diff
@@ -117,6 +121,7 @@ func TestConfigProvider_GetAzureClientConfig(t *testing.T) {
 			name: "Should return a ClientCertificateCredential",
 			config: config.AzureConfig{
 				Credentials: config.AzureClientOpt{
+					ClientCredentialsType:     config.AzureClientCredentialsTypeCertificate,
```
Comment (orestisfl) [inline resources/providers/azurelib/auth/credentials_test.go:124]: I would also add tests to cover the cases where credentials type is empty / unknown value.

Comment (moukoublen) [reply resources/providers/azurelib/auth/credentials_test.go:136]: The first test case has empty credentials type value.  I added one with `unknown client credentials type`.

File: resources/providers/azurelib/auth/credentials_test.go [hunk 2]
```diff
@@ -44,13 +44,26 @@ func TestConfigProvider_GetAzureClientConfig(t *testing.T) {
...
 		},
+		{
+			name: "Should return a DefaultAzureCredential on unknown client credentials type",
+			config: config.AzureConfig{
+				Credentials: config.AzureClientOpt{
+					ClientCredentialsType: "unknown",
+				},
+			},
+			authProviderInitFn: initDefaultCredentialsMock(nil),
+			want: &AzureFactoryConfig{
```
Comment (orestisfl) [inline resources/providers/azurelib/auth/credentials_test.go:55]: Sorry for being nitpicky but this should probably return an error, it signifies a mismatch between integration and cloubeat

Comment (moukoublen) [reply resources/providers/azurelib/auth/credentials_test.go]: Sure,  I added an error in that case. Not in an empty string, though. Empty string fallbacks to default.

**Review by orestisfl — APPROVED** [review]
:+1:
